src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,57 @@
+.calendar {
+  display: flex;
+  box-sizing: border-box;
+  max-width: $calendar-width;
+  min-width: $calendar-width;
+  padding: $calendar-paddings;
+  gap: $calendar-gap;
+  flex-wrap: wrap;
+
+  @each $key, $value in $day-of-week {
+    &--start-day-#{$key} {
+      .calendar__day:first-child {
+        margin-left: #{$value};
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+
+    font-family: Arial, sans-serif;
+    height: $day-height;
+    width: $day-width;
+    flex-shrink: 0;
+    background-color: $day-bg-color;
+
+    font-size: $day-font-size;
+
+    border: $day-border-width solid $day-border-color;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: "#{$day}";
+        line-height: $day-height;
+        width: $day-width;
+        text-align: center;
+      }
+    }
+
+    transition: all 0.5s;
```

Comment:
use transition for property or properties which should be changed, for example transition: `color $animation-time, width $animation-time`, try to avoid using `all`